warp/src/main/java/com/schibsted/nmp/warp/components/WarpModal.kt

```diff
@@ -0,0 +1,142 @@
+package com.schibsted.nmp.warp.components
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Close
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.window.Dialog
+import androidx.compose.ui.window.DialogProperties
+import com.schibsted.nmp.warp.theme.WarpTheme.colors
+import com.schibsted.nmp.warp.theme.WarpTheme.dimensions
+
+@Composable
+fun WarpModal(
+    title: String,
+    body: String,
+    onDismiss: () -> Unit,
+    subTitle: String? = null,
+    confirmButtonText: String? = null,
```

Comment:
I liked better the Android way with Dialogs, with a Primary and a Secondary buttons (which are what is used in the end)
and then according the names of the actions as onSecondary, onPrimary...
Granted then, dismiss has to be handled differently... 🤔

Reply:
Do you mean that cancel button should have an own onDismiss()?

Reply:
What I meant was that the names "confirm" and "cancel" are too specific to a usecase, while the modal could have actions that are neither "confirm" or "cancel"
Since your buttons are indeed primary and secondary, why not call the parameters the same way?
The point about onDismiss is that if you call the current "cancel" button "secondary", you imply that the use of this button might _not_ be to cancel the modal, so the dismissal would have to be handled differently by first disassociate these two.